### DISCUSSION TITLE
"error": "--to is not a valid sha pointer: \"HEAD\", --from is not a valid sha pointer: \"origin/develop\""

### DISCUSSION THREAD

**Discussion by niv-goldenberg** [scolladon/sfdx-git-delta#387] [Show and tell]
I'm opening this issue with a solution for documentation as I just spent several hours to solve it.

I had the same issue as here: https://bytemeta.vip/repo/scolladon/sfdx-git-delta/issues/373

The owner of the issue has stopped using Docker image as a solution. I propose a different solution that allows you to keep using the docker image.

I recently updated my docker images, which include all sfdx and sfdx-git-delta to the latest versions. After the update I got the  following issue: 
```
{
  "error": "--to is not a valid sha pointer: \"HEAD\", --from is not a valid sha pointer: \"origin/develop\"",
  "output": ".",
  "success": false,
  "warnings": []
}
```

Long story short, trying to access the git logs (by using git log) directly gave the solution of just applying the checkout (v3) working folder as a safe directory (before running any sfdx-git-delta commands), for example in my case:
`git config --global --add safe.directory /__w/core-platform/core-platform`

Hope it will save time for someone else in the future.

**Comment by scolladon** [comment]
Hi @niv-goldenberg !

Thanks for the time you spent on it and for the solution!

Can you give a bit more detail please ?
For example can you give the github action yaml file ?

I think we could convert this into a discussion if you agree

**Comment by packocz** [comment]
I'd also be very interested in more detail for the solution. I tried to apply it like below, but it's not helped:

```
deploy-dev:
        runs-on: 'ubuntu-latest'
        container: salesforce/salesforcedx:latest-rc-full
        #concurrency: deploy-dev
        env:
            SFDX_DISABLE_DNS_CHECK: true
        steps:
            - name: Clone Repository (Latest)
              uses: actions/checkout@v3
              with:
                  fetch-depth: 0 

            - name: 'Populate auth file with secret'
              shell: bash
              run: ......

            - name: 'Authenticate Org'
              run: ....

            - name: 'Set safe folder'
              run: git config --global --add safe.directory /__w/core-platform/core-platform

            - name: 'Install SFDX Git Delta plugin'
              run: echo y | sfdx plugins:install sfdx-git-delta

            - name: 'Generate Delta manifest'
              run: sfdx sgd:source:delta --from "successful-dev-build" --to "$GITHUB_SHA" --output .
```
Same command locally runs fine, only GitHub Action has the problem

**Comment by niv-goldenberg** [comment]
> I'd also be very interested in more detail for the solution. I tried to apply it like below, but it's not helped:
> 
> ```
> deploy-dev:
>         runs-on: 'ubuntu-latest'
>         container: salesforce/salesforcedx:latest-rc-full
>         #concurrency: deploy-dev
>         env:
>             SFDX_DISABLE_DNS_CHECK: true
>         steps:
>             - name: Clone Repository (Latest)
>               uses: actions/checkout@v3
>               with:
>                   fetch-depth: 0 
> 
>             - name: 'Populate auth file with secret'
>               shell: bash
>               run: ......
> 
>             - name: 'Authenticate Org'
>               run: ....
> 
>             - name: 'Set safe folder'
>               run: git config --global --add safe.directory /__w/core-platform/core-platform
> 
>             - name: 'Install SFDX Git Delta plugin'
>               run: echo y | sfdx plugins:install sfdx-git-delta
> 
>             - name: 'Generate Delta manifest'
>               run: sfdx sgd:source:delta --from "successful-dev-build" --to "$GITHUB_SHA" --output .
> ```
> 
> Same command locally runs fine, only GitHub Action has the problem

The only difference from what I have is the order of execution:
```
        # Bypass permission issue:
        git config --global --add safe.directory /__w/core-platform/core-platform

        # Run sf git delta:
        sfdx sgd:source:delta -t HEAD -f ${{ inputs.previous_commit_sha_alias }} -r . -o . --ignore .forceignore --loglevel debug
```

**Reply by packocz** [reply]
I failed to find the reason for the failure, but when I abandoned using the `salesforce/salesforcedx:latest-rc-full` docker image (as suggested in the related convo) in favour of the default ubuntu action container it started working fine!

**Comment by niv-goldenberg** [comment]
> 

Yes, I agree this should be converted into a discussion.

Thanks.

**Comment by jonny-harte** [comment]
Just come across the same issue. Trying to move to the docker image rather than installing node and SFDX each time but get the same error. Adding the safe.directory hasn't worked. Any idea how to fix this?

**Reply by scolladon** [reply]
I found something [here](https://salesforce.stackexchange.com/questions/379675/sfdx-plugins-are-not-getting-installed-via-docker)
It uses the official sfdx cli docker and tweak it a bit in order to install sfdx-cli by downloading it instead of using npm and also create some path.

What I would change in this dockerfile though is the installation channel.
It is using the `latest-rc` (release candidate) where I think a `latest` would be more stable.

I have not tried it personnally
Let us know if it helped

**Reply by niv-goldenberg** [reply]
> Just come across the same issue. Trying to move to the docker image rather than installing node and SFDX each time but get the same error. Adding the safe.directory hasn't worked. Any idea how to fix this?

Make sure the safe directory path is the correct one, using 'git log' command running on github actions.
